app/controllers/katello/concerns/smart_proxies_controller_extensions.rb

```diff
@@ -0,0 +1,45 @@
+module Katello
+  module Concerns
+    module SmartProxiesControllerExtensions
+      extend ActiveSupport::Concern
+
+      included do
+        append_view_path('app/views/foreman')
+        before_filter :find_resource_and_status, :only => [:pulp_storage, :pulp_status]
+        alias_method_chain :action_permission, :katello
+
+        def pulp_storage
```

Comment:
To me it would -- as I don't know why we would want to introduce some data as a UI only thing. Granted, I am also used to the Katello UI which inevitably requires APIs be built to support it (two birds one stone). If APIs are not in scope for this just let me know - some stories to track creating them would be nice.